src/app/views/authentication/Authentication.tsx

```diff
@@ -38,69 +44,112 @@ const Authentication = (props: any) => {
       const authResponse = await authenticationWrapper.logIn();
       if (authResponse) {
         setLoginInProgress(false);
-        dispatch(getAuthTokenSuccess(!!authResponse.accessToken))
-        dispatch(getConsentedScopesSuccess(authResponse.scopes))
+        dispatch(getAuthTokenSuccess(!!authResponse.accessToken));
+        dispatch(getConsentedScopesSuccess(authResponse.scopes));
       }
     } catch (error) {
       const { errorCode } = error;
-      dispatch(setQueryResponseStatus({
-        ok: false,
-        statusText: messages['Authentication failed'],
-        status: errorCode === 'popup_window_error'
-          ? translateMessage('popup blocked, allow pop-up windows in your browser')
-          : errorCode ? errorCode.replace('_', ' ') : '',
-        messageType: MessageBarType.error
-      }));
+      dispatch(
+        setQueryResponseStatus({
+          ok: false,
+          statusText: messages['Authentication failed'],
+          status: getErrorAndHint(errorCode),
+          messageType: MessageBarType.error,
+        })
+      );
       setLoginInProgress(false);
       telemetry.trackException(
         new Error(errorTypes.OPERATIONAL_ERROR),
         SeverityLevel.Error,
         {
           ComponentName: componentNames.AUTHENTICATION_ACTION,
-          Message: `Authentication failed: ${errorCode ? errorCode.replace('_', ' ') : ''}`,
-        });
+          Message: `Authentication failed: ${
+            errorCode ? errorCode.replace('_', ' ') : ''
+          }`,
+        }
+      );
     }
   };
 
-  const showProgressSpinner = (): React.ReactNode => {
-    return <div className={classes.spinnerContainer}>
-      <Spinner className={classes.spinner} size={SpinnerSize.medium} />
-      {!minimised && <Label>
-        <FormattedMessage id={`Signing you ${loginInProgress ? 'in' : 'out'}...`} />
-      </Label>}
-    </div>;
+  const getErrorAndHint = (errorCode: string): string => {
+    const errorMessageHint : string = translateMessage(errorCode);
+
+    for(const errorItem of AuthErrorList){
+      if(errorItem === errorCode){
+        authenticationWrapper.clearCache();
+        deleteHomeAccountId();
+        window.sessionStorage.clear();
+      }
+    }
+    return errorCode.replace('_', ' ') + '      ' + translateMessage('Tip') + ':  ' + errorMessageHint;
+  };
+
+  const deleteHomeAccountId = () : void => {
+    localStorage.removeItem(HOME_ACCOUNT_KEY);
   }
```

Comment:
This method has already been declared in the `modules\authentication\AuthenticationWrapper.ts` file as a private method. Changing it to a public function enables access to it from outside of the class

Reply:
Okay. Let me fix this also